File: pico-compare-c++/CPP files/E05-LCD-INPUT.cpp
#include <Arduino.h>
#include <LiquidCrystal_I2C.h>

// Initialize the LCD
LiquidCrystal_I2C lcd(0x27, 16, 2);

void setup() {
    // Initialize the LCD and Serial communication
    lcd.init();
    lcd.backlight();
    Serial.begin(115200);
    
    // Print ready message on LCD
    lcd.print("Ready...");
}

void loop() {
    if (Serial.available()) {
        // Read the input from Serial
        String inputData = Serial.readStringUntil('\n');
        
        // Remove any trailing newline or spaces
        inputData.trim(); 
        
        // Check if the input specifies the line
        if (inputData.startsWith("1:")) {
            // Clear the line
            lcd.setCursor(0, 0);
            lcd.print("                ");
            
            // # Write the message
            lcd.setCursor(0, 0);
            lcd.print(inputData.substring(2,18));

        } else if (inputData.startsWith("2:")) {
            // Clear the line
            lcd.setCursor(0, 1);
            lcd.print("                ");
            
            // Write the message
            lcd.setCursor(0, 1);
            lcd.print(inputData.substring(2,18));
            
        } else {
            // Invalid data feedback
            Serial.println("Invalid input. Use '1:message' or '2:message'.");
        }
    }
}

File: esp32-compare-micropython/E05_lcd_input.py
from machine import SoftI2C, Pin
from i2c_lcd import I2cLcd
import sys

# Initialize the LCD
i2c = SoftI2C(scl=Pin(22), sda=Pin(21), freq=400000)
lcd = I2cLcd(i2c, 0x27, 2, 16)

# Print ready message on LCD
lcd.putstr("Ready...")

while True:
    try:
        # Read input from REPL
        input_data = input()
        
        # Remove any trailing newline or spaces
        input_data = input_data.strip()

        # Check if the input specifies the line
        if input_data.startswith("1:"):
            # Clear the line
            lcd.move_to(0, 0)
            lcd.putstr(" " * 16)
            
            # Write the message
            lcd.move_to(0, 0)
            lcd.putstr(input_data[2:18])  
        
        elif input_data.startswith("2:"):
            # Clear the line
            lcd.move_to(0, 1)
            lcd.putstr(" " * 16)
            
            # Write the message
            lcd.move_to(0, 1)
            lcd.putstr(input_data[2:18])
            
        else:
            # Invalid data feedback
            print("Invalid input. Use '1:message' or '2:message'.")
            
    except KeyboardInterrupt:
        # Exit program on keyboard interrupt
        print("\nExiting...")
        sys.exit()